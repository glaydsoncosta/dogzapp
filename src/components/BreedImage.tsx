import React from 'react';
import { StyleSheet, Text, TouchableWithoutFeedback, View } from 'react-native';
import FastImage from 'react-native-fast-image';
import { heightPercentToDP, widthPercentToDP } from '../util/scale';

type BreedImageProps = {
  imageURL: string;
  onPress: () => void;
};

const DEFAULT_IMAGE_WIDTH = widthPercentToDP('10%');
const DEFAULT_IMAGE_HEIGHT = heightPercentToDP('20%');

export default function BreedImage(props: BreedImageProps) {
  const { imageURL, onPress } = props;

  return (
    <TouchableWithoutFeedback onPress={() => onPress()}>
      <View style={styles.mainContainer}>
        <FastImage source={{ uri: imageURL }} resizeMode={FastImage.resizeMode.cover} style={styles.imageStyle} />
      </View>
    </TouchableWithoutFeedback>
  );
}

const styles = StyleSheet.create({
  mainContainer: {
    width: DEFAULT_IMAGE_WIDTH,
    height: DEFAULT_IMAGE_HEIGHT,
    marginBottom: heightPercentToDP('1%'),
    paddingRight: widthPercentToDP('45%')
  },
  imageStyle: {
    width: 180,
    height: 180,
    borderRadius: 10
  }
});
