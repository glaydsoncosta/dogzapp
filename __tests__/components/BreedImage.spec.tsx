import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import BreedImage from '../../src/components/BreedImage';

test('Should render BreedImage component', () => {
  const snapShot = renderer.create(<BreedImage imageURL="https://images.dog.ceo/breeds/akita/512px-Ainu-Dog.jpg" onPress={() => {}} />).toJSON();
  expect(snapShot).toMatchSnapshot();
});
