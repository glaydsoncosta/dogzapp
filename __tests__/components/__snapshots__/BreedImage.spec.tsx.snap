// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render BreedImage component 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "height": 267,
      "marginBottom": 13.5,
      "paddingRight": 337.5,
      "width": 75,
    }
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "borderRadius": 10,
          "height": 180,
          "width": 180,
        },
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="cover"
      source={
        {
          "uri": "https://images.dog.ceo/breeds/akita/512px-Ainu-Dog.jpg",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;
